client/components/Planning/FeaturedPlanning/FeaturedPlanningModal.jsx

```diff
@@ -1,46 +1,375 @@
+                dirty: false,
+            });
+        } else if (this.props.loading && !nextProps.loading &&
+            nextProps.featuredPlanningItems.length === nextProps.total) {
+            // Loading complete
+            this.setState(this.getNewState(nextProps));
+        } else if (!this.props.loading) {
+            if (!isEqual(this.props.featuredPlanningItem, nextProps.featuredPlanningItem)) {
+                this.setState(this.getNewState(nextProps));
+            }
+
+            if (this.props.featuredPlanningItems !== nextProps.featuredPlanningItems) {
+                // on update notifications
+                const addedPlannings = difference(nextProps.featuredPlanIdsInList, this.props.featuredPlanIdsInList);
+                const removedPlannings = difference(this.props.featuredPlanIdsInList, nextProps.featuredPlanIdsInList);
+                const selectedPlanningIds = this.state.selectedPlanningIds.filter(
+                    (id) => nextProps.featuredPlanIdsInList.includes(id));
+                let unselectedPlanningIds = [...this.state.unselectedPlanningIds, ...addedPlannings];
+
+                unselectedPlanningIds = unselectedPlanningIds.filter(
+                    (id) => !removedPlannings.includes(id) && !selectedPlanningIds.includes(id));
+
+                const notifications = [], highlights = [...addedPlannings];
+
+                if (addedPlannings.length > 0) {
+                    addedPlannings.forEach((id) => {
+                        const item = nextProps.featuredPlanningItems.find((p) => p._id === id);
+
+                        notifications.push(`Story with slugline '${item.slugline}' is added to the list`);
+                    });
+                }
+
+                if (removedPlannings.length > 0) {
+                    removedPlannings.forEach((id) => {
+                        const item = this.props.featuredPlanningItems.find((p) => p._id === id);
+
+                        notifications.push(`Story with slugline '${item.slugline}' is removed from the list`);
+                    });
+                }
+
+                this.setState({
+                    unselectedPlanningIds: unselectedPlanningIds,
+                    selectedPlanningIds: selectedPlanningIds,
+                    dirty: xor(get(this.props, 'featuredPlanningItem.items', []), selectedPlanningIds).length > 0,
+                    notifications: notifications,
+                    highlights: highlights,
+                });
+            }
+        }
+    }
+
+    getNewState(props) {
+        return {
+            unselectedPlanningIds: difference(props.featuredPlanningItems.map((i) => i._id),
+                get(props, 'featuredPlanningItem.items', [])),
+            selectedPlanningIds: get(props, 'featuredPlanningItem.items', []),
+            dirty: false,
+            highlights: [],
+        };
+    }
+
+    onDateChange(date) {
+        this.props.loadFeaturedPlanningsData(date);
```

Comment:
Make sure to use the configured defaultTimezone when selecting the date to show, as the date could change based on the client browsers timezone. I figure that this entire feature will be based on `config.defaultTimezone`.

It might be good if `<DateInputPopup>` had a timezone attribute that could control this.